services/user_service.go

```diff
@@ -5,6 +5,23 @@ import (
 	"golang-microservices-mvc/utils"
 )
 
-func GetUserById(userId int64) (*domain.User, *utils.ApplicationError) {
-	return domain.GetUserById(userId)
+type (
+	UserServiceInterface interface {
+		GetUserById(int64) (*domain.User, *utils.ApplicationError)
+	}
+	userService struct {
+	}
+)
+
+var (
+	UserService UserServiceInterface
```

Comment:
I don't actually like creating an empty interface and then assigning a struct instance to it. You could just create the UserService:
```Go
var UserService = &userService{}
```
Later you can use it whenever as a `UserServiceInterface` as it implement this interface.

Reply:
Done, please have a look